crates/miden-tx/src/host/link_map.rs

```diff
@@ -0,0 +1,292 @@
+use core::cmp::Ordering;
+
+use miden_objects::{Felt, Word, assembly::mast::MastNodeExt};
+use vm_processor::{
+    AdviceProvider, AdviceSource, ContextId, ErrorContext, ExecutionError, ProcessState,
+};
+
+/// A map based on a sorted linked list.
+///
+/// This type enables access to the list in kernel memory.
+///
+/// See link_map.masm for docs.
+///
+/// # Warning
+///
+/// The functions on this type assume that the provided map_ptr points to a valid map in the
+/// provided process. If those assumptions are violated, the functions may panic.
+#[derive(Debug, Clone, Copy)]
+pub struct LinkMap<'process> {
+    map_ptr: u32,
+    process: ProcessState<'process>,
+}
```

Comment:
This is fine for now, but another option could be to have a parallel structure on the Rust side to take entries in MASM. It may be more difficult to keep the in sync, but would be a bit more efficient.